operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -215,6 +215,7 @@ private String getLockName(String namespace, String name) {
                     Set<Condition> conditions = validate(cr);
                     conditions.add(StatusUtils.getPausedCondition());
                     status.setConditions(new ArrayList<>(conditions));
+                    status.setObservedGeneration(cr.getMetadata().getGeneration());
```

Comment:
Are we sure that this is the observed generation? Shouldn't we here take the observed generation from the previous status?